src/Facility.Core/Http/HttpClientService.cs

```diff
@@ -101,8 +103,66 @@ protected HttpClientService(HttpClientServiceSettings? settings, Uri? defaultBas
 			{
 				var contentType = mapping.RequestBodyContentType ?? requestHeaders?.GetContentType();
 				httpRequest.Content = GetHttpContentSerializer(requestBody.GetType()).CreateHttpContent(requestBody, contentType);
-				if (m_disableChunkedTransfer)
+				if (m_shouldCompressRequest(request))
+					httpRequest.Content = await CompressContentAsync(httpRequest.Content, cancellationToken).ConfigureAwait(false);
+				else if (m_disableChunkedTransfer)
```

Comment:
```suggestion
				if (m_disableChunkedTransfer)
```
Why `else`?

Reply:
I was thinking that creating a new HttpContent from a MemoryStream would just naturally have the side-effect of loading into a buffer (and this call would be redundant), but perhaps HttpRequestMessage doesn't see it that way.

Reply:
I guess I was also anticipating potential changes to the implementation of `CompressContentAsync`, in which case we might forget to drop this `else`.